app/controllers/plans_controller.rb

```diff
@@ -125,16 +123,18 @@ def to_entity(hash)
   # filters
 
   def do_jwt_authentication!
-    JWTUtils.decode(params[:token], external_plan_service[:jwt_secret]).on_error {
+    plans_api.authorize(params[:token]).on_error {
       logger.error("Unauthorized", nil, token: params[:token])
       render json: {error: :unauthorized}, status: 401
     }
   end
 
   def ensure_external_plan_service_in_use!
-    unless external_plan_service[:active]
-      raise ActiveRecord::RecordNotFound
-    end
+    raise ActiveRecord::RecordNotFound unless plans_api.active?
```

Comment:
Ok. This doesn't feel like very obvious way to do that because it doesn't communicate the intent. Also, wouldn't this be something that can change?